dlc/src/lib.rs

```diff
@@ -0,0 +1,703 @@
+            .unwrap()
+            * fee_rate;
+        let cet_fee = u64::try_from(CET_BASE_WEIGHT + self.final_address.script_pubkey().len() * 4)
+            .unwrap()
+            * fee_rate;
+        let fund_out = self.collateral + fund_fee + cet_fee;
+        if self.input_amount < fund_out {
+            return Err(Error::InvalidArgument);
+        }
+
+        let change_output = TxOut {
+            value: self.input_amount - fund_out,
+            script_pubkey: self.change_address.script_pubkey(),
+        };
+
+        Ok((change_output, fund_fee, cet_fee))
+    }
+
+    fn get_tx_input_info(&self, sequence: u32) -> Vec<TxIn> {
+        self.inputs
+            .iter()
+            .map(|i| TxIn {
+                previous_output: i.outpoint,
+                script_sig: Script::new(),
+                sequence,
+                witness: Vec::new(),
+            })
+            .collect()
+    }
+}
+
+/// Create the transactions for a DLC contract based on the provided parameters
+pub fn create_dlc_transactions(
+    local_params: &PartyParams,
+    remote_params: &PartyParams,
+    outcomes: &[Outcome],
+    refund_lock_time: u32,
+    fee_rate: u64,
+    lock_time: u32,
+) -> Result<DlcTransactions, Error> {
+    let total_collateral = local_params.collateral + remote_params.collateral;
+
+    let has_proper_outcomes = outcomes
+        .iter()
+        .all(|o| o.local + o.remote == total_collateral);
+
+    if !has_proper_outcomes {
+        return Err(Error::InvalidArgument);
+    }
+
+    let (local_change_output, local_fund_fee, local_cet_fee) =
+        local_params.get_change_output_and_fees(fee_rate)?;
+    let (remote_change_output, remote_fund_fee, remote_cet_fee) =
+        remote_params.get_change_output_and_fees(fee_rate)?;
+
+    let fund_output_value = local_params.input_amount + remote_params.input_amount
+        - local_change_output.value
+        - remote_change_output.value
+        - local_fund_fee
+        - remote_fund_fee;
+
+    assert_eq!(
+        total_collateral + local_cet_fee + remote_cet_fee,
+        fund_output_value
+    );
+
+    let sequence = if lock_time == 0 {
+        DISABLE_LOCKTIME
+    } else {
+        ENABLE_LOCKTIME
+    };
+    let local_tx_ins = local_params.get_tx_input_info(sequence);
+    let remote_tx_ins = remote_params.get_tx_input_info(sequence);
+
+    let fund_tx = create_funding_transaction(
+        &local_params.fund_pubkey,
+        &remote_params.fund_pubkey,
+        fund_output_value,
+        &local_tx_ins,
+        &remote_tx_ins,
+        local_change_output,
+        remote_change_output,
+        lock_time,
+    );
+
+    let fund_outpoint = OutPoint {
+        txid: fund_tx.txid(),
+        vout: 0,
+    };
+
+    let fund_tx_in = TxIn {
+        previous_output: fund_outpoint,
+        witness: Vec::new(),
+        script_sig: Script::new(),
+        sequence: 0xffffffff,
```

Comment:
Yeah there was definitely a bug here thanks! I would like to add something in the integration tests to check this, but since I'm using the same nLockTime for fund tx and cet I'm not sure how to do it. Would it make sense to have two parameters?

Reply:
I would lean towards two parameters unless it's causing other issue ?

Reply:
Two parameters added (https://github.com/p2pderivatives/rust-dlc/pull/2#discussion_r487647188) and both tested now